components/dashboard/src/service/service.tsx

```diff
@@ -51,6 +54,23 @@ function createGitpodService<C extends GitpodClient, S extends GitpodServer>() {
     return new GitpodServiceImpl<C, S>(proxy, { onReconnect });
 }
 
+function instrumentWebSocketConnection(connectionProvider: WebSocketConnectionProvider): void {
+    const originalCreateWebSocket = connectionProvider["createWebSocket"];
+    connectionProvider["createWebSocket"] = (url: string) => {
+        return originalCreateWebSocket.call(
+            connectionProvider,
+            url,
+            new Proxy(WebSocket, {
```

Comment:
we have to instrument the underlying web socket, not reconnecting, because reconneting never fires error or close and it will disturb data